### ISSUE TITLE
Snap issue solved by installing .deb

### ISSUE THREAD

**Issue by hollowdoor** [snapcrafters/atom#58]
I have an issue for Atom that only effects a snap install, or just me (I'm not sure). But the problem is resolved by using the official .deb instead. https://github.com/atom/atom/issues/20937

**Comment by JonnyB247** [comment]
I have the same problem on both my computers running the snap version on Ubuntu 20.04 x64. It happened after upgrading Atom to 1.48.0.

Starting Atom from the terminal instead of from a launcher seems to be a workaround.

**Comment by Danfro** [comment]
Yes running atom from terminal does avoid this snap-issue. Good to know. Thanks @JonnyB247 for sharing.

**Comment by hollowdoor** [comment]
The problem is described here: https://github.com/atom/atom/issues/20937#issuecomment-645233952

**Comment by pwaring** [comment]
Same issue on 20.04 here after I started using the Snap yesterday to avoid having to manually download the .deb from the Atom website every time there's a new version. For now I've had to switch back to using the .deb.

**Comment by lokal-profil** [comment]
as a quick hack edit `/var/lib/snapd/desktop/applications/atom_atom.desktop` and nuke the "ATOM_DISABLE_SHELLING_OUT_FOR_ENVIRONMENT=false /usr/bin/atom" bit

**Comment by Danfro** [comment]
Cool. Thanks for that hack. Does the job for now.

**Comment by diwic** [comment]
I think the line should look like this:

 > Exec=env BAMF_DESKTOP_FILE_HINT=/var/lib/snapd/desktop/applications/atom_atom.desktop ATOM_DISABLE_SHELLING_OUT_FOR_ENVIRONMENT=false /snap/bin/atom %F

Because `ATOM_DISABLE_SHELLING_OUT_FOR_ENVIRONMENT=false` is from [upstream](https://github.com/atom/atom/blob/master/resources/linux/atom.desktop.in), it should probably be preserved.

Looking at the git history for this project, it seems like `snapcraftctl build` was added just a few days ago. Maybe that's what messes up the `Exec=` line?

**Comment by valerauko** [comment]
Same kind of limitation happens with #20 (broken with snap, works with deb)
